src/md/compiler/import.cpp

```diff
@@ -1342,19 +1327,17 @@ STDMETHODIMP RegMeta::EnumProperties(         // S_OK, S_FALSE, or error.
         // instantiating a new ENUM
         CMiniMdRW       *pMiniMd = &(m_pStgdb->m_MiniMd);
         RID         ridPropertyMap;
-        PropertyMapRec *pPropertyMapRec;
 
         // get the starting/ending rid of properties of this typedef
         IfFailGo(pMiniMd->FindPropertyMapFor(RidFromToken(td), &ridPropertyMap));
         if (!InvalidRid(ridPropertyMap))
         {
-            IfFailGo(m_pStgdb->m_MiniMd.GetPropertyMapRecord(ridPropertyMap, &pPropertyMapRec));
-            ridStart = pMiniMd->getPropertyListOfPropertyMap(pPropertyMapRec);
+            IfFailGo(pMiniMd->getStartPropertyListOfPropertyMap(ridPropertyMap, &ridStart));
             IfFailGo(pMiniMd->getEndPropertyListOfPropertyMap(ridPropertyMap, &ridEnd));
-            ridMax = pMiniMd->getCountPropertys() + 1;
-            if(ridStart == 0) ridStart = 1;
-            if(ridEnd > ridMax) ridEnd = ridMax;
-            if(ridStart > ridEnd) ridStart=ridEnd;
+            //ridMax = pMiniMd->getCountPropertys() + 1;
```

Comment:
This is a bunch of suspicious code I'll need to investigate more. Hence the WIP.

Reply:
Welp, as expected, this code predates TFS source history of the CLR. It kind of looks like it was trying to fix some bad tooling. This code would no longer work after the change that prevents `ridStart` from ever be reported as zero, so I propose we just delete this (and other places like this).